### PR TITLE
Team Batting (FanGraphs) => Add unit tests; use new consolidated data source

### PR DESCRIPTION
**Pull request by TheCleric**
This PR is the next step for #91 , now for Team Batting.

In this PR we've added unit tests, and cut team_batting to using the new shared method for getting FanGraphs data.

I've also removed the extraneous parameters in the FanGraphs datasource functions that are now auto-determined.

I've also added a MyPy checker to CI that will NOT fail the PR, but just so we can slowly ensure data types are being respected (could catch potential bugs).

### CODE REVIEW

**Comment by schorrm** [comment]
Does this dramatically change the number of columns, though?

**Comment by schorrm** [comment]
Never mind, that's a dumb comment, I got this confused with the regular batting.

File: pybaseball/datasources/fangraphs.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 ROOT_URL = 'https://www.fangraphs.com'
 
 
-def get_fangraphs_tabular_data_from_html(html: str, xpath: str, percentage_columns: List[str] = None, non_numeric_columns: List[str] = None) -> pd.DataFrame:
+def get_fangraphs_tabular_data_from_html(html: Union[str, bytes], xpath: str) -> pd.DataFrame:
```
Comment (schorrm) [inline pybaseball/datasources/fangraphs.py:12]: <3

File: pybaseball/team_batting.py [hunk 2]
```diff
@@ -1,79 +1,36 @@
...
-            data[col] = data[col].astype(float)/100.
-        else:
-            pass
-
-    # convert columns to numeric
-    not_numeric = ['Team']
-    numeric_cols = [col for col in data.columns if col not in not_numeric]
-    data[numeric_cols] = data[numeric_cols].astype(float)
-    return data
-
```
Comment (schorrm) [inline pybaseball/team_batting.py:56]: This is some sweet sweet dividends here

File: pybaseball/team_fielding.py [hunk 3]
```diff
@@ -30,9 +30,7 @@ def team_fielding(start_season, end_season=None, league='all', ind=1):
 
     fg_data = fangraphs.get_fangraphs_tabular_data_from_url(
         _FG_TEAM_FIELDING_URL.format(start_season=start_season, end_season=end_season, league=league, ind=ind),
-        '//table[@class="rgMasterTable"]',
-        percentage_columns=['CS%', 'lgCS%'],
-        non_numeric_columns=['Team', 'Name']
+        '//table[@class="rgMasterTable"]'
```
Comment (schorrm) [inline pybaseball/team_fielding.py:33]: Substantive: is this always "rgMasterTable"? Can we streamline this call?

File: tests/pybaseball/datasources/test_fangraphs.py [hunk 4]
```diff
@@ -60,9 +60,7 @@ class TestDatasourceFangraphs:
     def test_get_table(self, sample_html, sample_processed_result):
         actual_result = fangraphs.get_fangraphs_tabular_data_from_html(
             sample_html,
-            '//table',
-            ["CS%"],
-            ["Team"]
+            '//table'
```
Comment (schorrm) [inline tests/pybaseball/datasources/test_fangraphs.py:63]: Is it only ever something else in the testing framework? Can we default it to rg and have the `//table` only in the tests? wondering

File: pybaseball/team_fielding.py (hunk 3 above)
Comment (TheCleric) [reply pybaseball/team_fielding.py:33]: Great question. Let me check.

File: tests/pybaseball/datasources/test_fangraphs.py (hunk 4 above)
Comment (TheCleric) [reply tests/pybaseball/datasources/test_fangraphs.py:63]: If so I can just adjust the HTML of the test to add that class.

File: pybaseball/team_batting.py (hunk 2 above)
Comment (TheCleric) [reply pybaseball/team_batting.py:56]: I always love deleting code more than adding code.

File: pybaseball/team_fielding.py (hunk 3 above)
Comment (TheCleric) [reply pybaseball/team_fielding.py:33]: So here is all the places we'd end up implementing this in the future:

- batting_leaders.py
- pitching_leaders.py
- team_pitching.py

All of them use this table class. So I could have this as a default param. I'll modify it shortly.

File: tests/pybaseball/datasources/test_fangraphs.py (hunk 4 above)
Comment (TheCleric) [reply tests/pybaseball/datasources/test_fangraphs.py]: Or not because it's actual FG HTML, so it's already there. lol

**Comment by TheCleric** [comment]
Updated with suggestions.

**Comment by schorrm** [comment]
Looking good. Wonderful work, as usual.

**Pull request merged**